cmd/bosun/sched/notify.go

```diff
@@ -209,3 +211,51 @@ func (s *Schedule) AddNotification(ak expr.AlertKey, n *conf.Notification, start
 	}
 	s.Notifications[ak][n.Name] = started
 }
+
+type actionNotificationData struct {
+	States     []*State
+	User       string
+	Message    string
+	ActionType string
+}
+
+var actionNotificationSubjectTemplate *template.Template
+var actionNotificationBodyTemplate *template.Template
+
+func init() {
+	subject := `{{$first := index .States 0}}{{$count := len .States}}
+{{.User}} {{.ActionType}}
+{{if gt $count 1}} {{$count}} Alerts. 
+{{else}} Incident #{{$first.Last.IncidentId}} ({{$first.Subject}}) 
+{{end}}`
+	body := `{{$count := len .States}}{{.User}} {{.ActionType}} {{$count}} alert{{if gt $count 1}}s{{end}}:
+<strong>Message:</strong> {{.Message}}
+<strong>Incidents:</strong>
+<ul>{{range .States}}
+<li><a href="/incident?id={{.Last.IncidentId}}">#{{.Last.IncidentId}}:<a/> {{.Subject}}</li>{{end}}
+</ul>`
+	actionNotificationSubjectTemplate = template.Must(template.New("").Parse(strings.Replace(subject, "\n", "", -1)))
```

Comment:
Can this use a similar pattern for the unknownTemplate, which has a default but can be set by a user?